tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers.ts

```diff
@@ -469,6 +478,37 @@ const reducer = createReducer(
       return newState;
     }
   ),
+  on(
+    actions.sourceFileListRequested,
+    (state: DebuggerState): DebuggerState => {
+      return {
+        ...state,
+        sourceCode: {
+          ...state.sourceCode,
+          sourceFileListLoaded: {
+            ...state.sourceCode.sourceFileListLoaded,
+            state: DataLoadState.LOADING,
+          },
+        },
+      };
+    }
+  ),
+  on(
+    actions.sourceFileListLoaded,
+    (state: DebuggerState, sourceFileList): DebuggerState => {
+      return {
+        ...state,
+        sourceCode: {
+          ...state.sourceCode,
+          sourceFileListLoaded: {
+            state: DataLoadState.LOADED,
+            lastLoadedTimeInMs: Date.now(),
+          },
+          sourceFileList: sourceFileList.sourceFiles,
```

Comment:
Based on the data access, we always pass a `run` when requesting sourceFiles.

To double check: is there any use case in the UX when a user wants to keep seeing stale source files?  or source files from a previously selected run?  (e.g. should we just merge the new + previous list of source files?)

I'd imagine if there's a run-selector UI control, a user might switch back and forth between 2 runs many times, and since we don't index the sourceFiles by run-id, we would need to re-fetch on each switch.

Reply:
Good question. The debugger v2 FE currently is implemented in a way that shows only one run at a time. This keeps the data structure simpler than what it would be if multiple runs are handled. This is sufficient IMO because when during TF debugging, you typically examine one run (i.e., one training job) at a time. There is less concern for comparing, say the loss curves from different hyperparameters from different runs, than in other TB dashboards such as scalars.